### ISSUE TITLE
OneClassPerFile clash with anonymous classes

### ISSUE THREAD

**Issue by JeroenDeDauw** [squizlabs/PHP_CodeSniffer#955]
I'm using the `OneClassPerFile` sniff in a PHP 7 codebase with usage of [anonymous classes](http://php.net/manual/en/language.oop5.anonymous.php). The sniff fails

```
FILE: ...gFrontend/tests/Integration/Validation/DonationValidatorTest.php

 123 | ERROR   | Only one class is allowed in a file
     |         | (Generic.Files.OneClassPerFile.MultipleFound)
 153 | WARNING | Duplicate class name
     |         | "WMDE\Fundraising\Tests\Integration\Validation\PaymentMethod"
     |         | found; first defined inDonationValidatorTest
     |         | /home/j/workspace/FundraisingFrontend/tests/Integration/Validation/DonationValidatorTest.php
     |         | on line 123 (Generic.Classes.DuplicateClassName.Found)
 153 | ERROR   | Only one class is allowed in a file
     |         | (Generic.Files.OneClassPerFile.MultipleFound)
 153 | ERROR   | Each class must be in a file by itself
     |         | (PSR1.Classes.ClassDeclaration.MultipleClasses)
----------------------------------------------------------------------
```

Relevant code

```php
	public function testNoPaymentTypeGiven_validatorReturnsFalse() {
		$donation = $this->newDonationWithPaymentMethod( new class implements PaymentMethod {
			public function getType(): string {
				return '';
			}
		} );

		$this->assertFalse( $this->donationValidator->validate( $donation )->isSuccessful() );

		$this->assertConstraintWasViolated(
			$this->donationValidator->validate( $donation ),
			'zahlweise'
		);
	}
```

Given the nature of anonymous classes, this restriction makes no sense. Therefore either making the sniff ignore anon classes, or adding an option that allows doing so, would be great.

> PHP_CodeSniffer version 1.5.5 (stable) by Squiz (http://www.squiz.net)

**Comment by gsherwood** [comment]
You're on an old unsupported branch of PHP_CodeSniffer so you don't have any of the PHP7-specific features, including anon class support.

To get support for anon classes, you'll need to upgrade to at least PHP_CodeSniffer 2.4.0 (released Nov 2015) although I'd strongly recommend moving to the latest 2.6.0.

I've tested this code on both 1.5.5 and 2.6.0 and can confirm the issue exists in 1.5.5 and not in 2.6.0. So closing this issue as it has been fixed in the 2.x branch.
